File: mobileBE/src/main/java/com/uth/mobileBE/dto/response/FeeInvoiceResponse.java
package com.uth.mobileBE.dto.response;

import com.uth.mobileBE.models.enums.PaymentMethod;
import com.uth.mobileBE.models.enums.StatusFeeInvoice;
import com.uth.mobileBE.models.enums.TypeFeeInvoice;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class FeeInvoiceResponse {
    private Long invoiceId;
    private Long libraryId;
    private Long readerId;
    private String readerName;
    private Long loanDetailId;

    private TypeFeeInvoice type;
    private PaymentMethod paymentMethod;
    private Double totalAmount;
    private StatusFeeInvoice status;

    private LocalDateTime createdAt;
    private LocalDateTime updateAt;
}

File: app/src/main/java/com/example/quanlythuvien/data/model/response/FeeInvoiceResponse.kt
package com.example.quanlythuvien.data.model.response

import com.google.gson.annotations.SerializedName

data class FeeInvoiceResponse(
    @SerializedName("invoiceId")
    val invoiceId: Long,

    @SerializedName("libraryId")
    val libraryId: Long,

    @SerializedName("readerId")
    val readerId: Long,

    @SerializedName("loanId")
    val loanId: Long?,

    @SerializedName("type")
    val type: String,

    @SerializedName("totalAmount")
    val totalAmount: Double,

    @SerializedName("status")
    val status: String,

    @SerializedName("createdAt")
    val createdAt: String,

    @SerializedName("updateAt")
    val updateAt: String?,

    @SerializedName("readerName")
    val readerName: String,

    @SerializedName("paymentMethod")
    val paymentMethod: String? = null,

    @SerializedName("description")
    val description: String? = null
) {
    // Các trường tính toán để tương thích với code cũ
    val invoiceType: String
        get() = type

    val amount: Double
        get() = totalAmount

    val updatedAt: String?
        get() = updateAt
}